thread.c

```diff
@@ -4287,8 +4357,16 @@ select_single_cleanup(VALUE ptr)
 }
 
 int
-rb_wait_for_single_fd(int fd, int events, struct timeval *tv)
+rb_wait_for_single_fd(int fd, int events, struct timeval *timeout)
 {
+    VALUE scheduler = rb_current_thread_scheduler();
+    if (scheduler != Qnil) {
+        VALUE result = rb_funcall(scheduler, id_wait_for_single_fd, 3, INT2NUM(fd), INT2NUM(events),
+            rb_float_new((double)timeout->tv_sec + (0.000001f * timeout->tv_usec))
```

Comment:
Do we have method for doing this? Can we convert timeval into duration more easily?